Samples/NotifyUsers/AppBackend/Controllers/NotificationsController.cs

```diff
@@ -57,13 +54,15 @@ public async Task<HttpResponseMessage> Post(string pns, [FromBody]string message
                     break;
             }
 
-            if (outcome != null)
+            if (outcome == null)
             {
-                if (!((outcome.State == Microsoft.Azure.NotificationHubs.NotificationOutcomeState.Abandoned) ||
-                    (outcome.State == Microsoft.Azure.NotificationHubs.NotificationOutcomeState.Unknown)))
-                {
-                    ret = HttpStatusCode.OK;
-                }
+                return Request.CreateResponse(ret);
+            }
+
+            if (!(outcome.State == Microsoft.Azure.NotificationHubs.NotificationOutcomeState.Abandoned ||
```

Comment:
It's more of a personal preference but I would have moved the `||` operator to the beginning of the next line, something like this:
```
if (!((outcome.State == Microsoft.Azure.NotificationHubs.NotificationOutcomeState.Abandoned) 
    || (outcome.State == Microsoft.Azure.NotificationHubs.NotificationOutcomeState.Unknown)))
```
And it seems that this statement might be reversed to get rid of the `!` at the beginning and consequential parenthesis pair, but it's only a guess that might be incorrect. 🙂